src/psyclone/psyGen.py

```diff
@@ -2542,9 +2564,26 @@ def update(self):
         enddir = Comment(FortranStringReader("!$omp end parallel",
                                              ignore_comments=False))
         self.ast_end = enddir
-        # If end_idx+1 takes us beyond the range of the list then the
-        # element is appended to the list
-        self._parent.ast.content.insert(end_idx+1, enddir)
+        # FIXME: In case of:
```

Comment:
Yes, from memory it's all related to the AST. Since (afaik) this is only temporary till the PSyIR is fully supported  - tb(very)h I don't care, I would leave it in till the whole update functions can be removed (add a #todo to a psyir ticket ).

Reply:
I believe that should be #435, which I have added (#500 is use SymbolTable).

Reply:
Oh yes, thanks.